spark/v3.1/spark/src/test/java/com/netease/arctic/spark/SparkTestContext.java

```diff
@@ -450,4 +457,30 @@ public void assertHiveDesc(List<Object[]> rows, List<String> cols, List<String>
         descPartitionKey.stream().sorted().distinct().toArray());
   }
 
+  public void testWrite(ArcticTable table, LocationKind locationKind, List<Record> records) throws IOException {
+    AdaptHiveGenericTaskWriterBuilder builder = AdaptHiveGenericTaskWriterBuilder
+        .builderFor(table)
+        .withTransactionId(1);
+
+    TaskWriter<Record> changeWrite = builder.buildWriter(locationKind);
+    for (Record record: records) {
+      changeWrite.write(record);
+    }
+    DataFile[] dataFiles = changeWrite.complete().dataFiles();
+    if(table.isKeyedTable()) {
+      KeyedTable keyedTable = table.asKeyedTable();
+      OverwriteBaseFiles overwriteBaseFiles = keyedTable.newOverwriteBaseFiles();
+      Arrays.stream(dataFiles).forEach(overwriteBaseFiles::addFile);
+      overwriteBaseFiles.withTransactionId(keyedTable.beginTransaction(System.currentTimeMillis() + ""));
+      overwriteBaseFiles.commit();
+    } else if (table.isUnkeyedTable()) {
+      UnkeyedTable unkeyedTable = table.asUnkeyedTable();
+      OverwriteFiles overwriteFiles = unkeyedTable.newOverwrite();
+      Arrays.stream(dataFiles).forEach(overwriteFiles::addFile);
+      overwriteFiles.commit();
+    }else {
+      throw new IllegalStateException("Table is neither keyed nor unkeyed");
+    }
+  }
```

Comment:
adapt hive write should in hive test context.   
all common method about hive should be in test context